### PR TITLE
Raise an exception when required configuration is missing

### PR DESCRIPTION
**Pull request by xoen**
Replaced `os.environ.get()` call with `os.environ[]` when there is
no sensible default and the user should configure a setting.

**NOTE**: Tests pass and `make run` seems to work but I'm not 100% this
is not breaking something because I need to set these variable in some
other places.

### CODE REVIEW

File: controlpanel/settings/common.py [hunk 1]
```diff
@@ -252,9 +252,9 @@
 DATABASES = {
     "default": {
         "ENGINE": "django.db.backends.postgresql",
-        "NAME": os.environ.get("DB_NAME", PROJECT_NAME),
```
Comment (andyhd) [inline controlpanel/settings/common.py]: This one has a default

File: controlpanel/settings/common.py [hunk 2]
```diff
@@ -252,9 +252,9 @@
...
-        "USER": os.environ.get("DB_USER", ""),
-        "PASSWORD": os.environ.get("DB_PASSWORD", ""),
```
Comment (andyhd) [inline controlpanel/settings/common.py]: These should probably have defaults of `PROJECT_NAME`

**Comment by xoen** [comment]
@andyhd I've changed this but I'm not 100% sure it doesn't break anything unfortunately.

`make test` runs and pass locally and AFAIK all these environment variables involved are set by the helm chart - which is the good news. The bad news is that probably devs will have to add more values to their `.env` file.

Could you have a look and possibly check this branch and see if you see anything wrong with it? Thanks 😌

**Comment by andyhd** [comment]
Ah, now I remember why they have defaults - `collectstatic` loads the settings (but it doesn't need them). Maybe there should be some overrides just for collectstatic.

File: controlpanel/settings/common.py (hunk 1 above)
Comment (xoen) [reply controlpanel/settings/common.py]: I know but we pass them anyway and I think we should try and be explicit as we have to pass the rest anyway.

**Comment by xoen** [comment]
I see the `docker build` fails (on CircleCI), if `collectstatic` doesn't use them, maybe I can just pass an emtpy string, it would be a bit 💩 - but I'll play with it and see how it goes.

**Comment by xoen** [comment]
Right. I'm going to abort this as it was more challenging than imagined especially for the benefits of doing this 😭

**Pull request closed without merging**